File: examples-sdk/python/032-basic-failover-ha.py
#!/usr/bin/env python3
"""032-basic-failover-ha — Python SDK equivalent
Compile: vil compile --from python --input 032-basic-failover-ha.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("payment-gateway-ha", port=8080)
primary = server.service_process("primary")
primary.endpoint("GET", "/health", "primary_health")
primary.endpoint("POST", "/charge", "primary_charge")
backup = server.service_process("backup")
backup.endpoint("GET", "/health", "backup_health")
backup.endpoint("POST", "/charge", "backup_charge")
server.compile()

File: examples-sdk/typescript/032-basic-failover-ha.ts
#!/usr/bin/env tsx
// 032-basic-failover-ha — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 032-basic-failover-ha.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("payment-gateway-ha", 8080);
const primary = new ServiceProcess("primary");
primary.endpoint("GET", "/health", "primary_health");
primary.endpoint("POST", "/charge", "primary_charge");
server.service(primary);
const backup = new ServiceProcess("backup");
backup.endpoint("GET", "/health", "backup_health");
backup.endpoint("POST", "/charge", "backup_charge");
server.service(backup);
server.compile();
